### ISSUE TITLE
Error Report 2018.05.03

### ISSUE THREAD

**Issue by kraftwerk15** [DynamoDS/Dynamo#8822]
## Dynamo version

2.0.0.??? Not Daily, latest stable.

## Operating system

Windows 7

## What did you do? 

Click on the Dynamo button under the Revit > Manage tab.

## What did you expect to see?

A functioning Dynamo.

## What did you see instead?

A non-functioning Dynamo. ;)
Seeing the below image instead.

## Explanation:
I have loaded in a custom library after starting 2.0 Dynamo. It will function normally while in that session. After closing Dynamo and attempting to start again, the below error message will display.

![image](https://user-images.githubusercontent.com/30668448/39579856-581154ac-4ead-11e8-9fa8-10e69897bc58.png)

Closing the application, starting the application again, then starting Dynamo will yield the following error message.

![dynocrash](https://user-images.githubusercontent.com/30668448/39580461-db39518a-4eae-11e8-9e76-285549d50d3f.PNG)

The .dll is my library that was working in 1.3.1 and earlier in 2.0, now no longer working.

**Comment by jnealb** [comment]
@mjkkirschner Any thoughts?

**Comment by mjkkirschner** [comment]
not really ;) - Does this happen with any other node library?
Is your dll a NodeModel library or just zero touch c# nodes?
Do you use any dynamo attributes in your library?
Do you use google analytics in your library?
Can you share your source?

**Comment by kraftwerk15** [comment]
No. Just my library. ;) 

I was having a tremendous amounts of issues trying to download archi-lab, until he made modifications for 2.0. Usually when I would try to download just his package, the Revit application would crash. Package Manager would download to the location, crash, then delete the files on crash.... with just archi-lab. 

My library was just zero-touch. Needed something along the lines of a dropdown, so added in CoreNodeModels. Even if I do not compile the new .cs file that is using some of the new stuff, I still have the same issues above. In 1.3, I could overwrite the .dll in the package location, relaunch Revit/Dynamo and be good to go. Now, I have to delete the package every time I need to make a modification to the .dll, then publish the package locally to be able to see the change. Rinse and repeat for every change.

No Google Analytics.

Here's a list of the new .cs that is causing trouble.
https://gist.github.com/kraftwerk15/fa88be48fd0d98f445bc1564fa224da9

Could this be that I am using a reference from the wrong location? 1.3 vs. 2.0?

**Comment by mjkkirschner** [comment]
I'm not sure from the information given, but you should not bundle any dynamo dlls with your package. It's quite possible you've bundled 1.3 dynamo core dlls and they are loaded along with the dynamo 2.0 dlls and chaos ensues.

can you show your package bin folder?

**Comment by kraftwerk15** [comment]
Top Level:

![image](https://user-images.githubusercontent.com/30668448/39596905-73dcfad4-4ed9-11e8-9ca7-3c28984a70fe.png)

BIN Debug:

![image](https://user-images.githubusercontent.com/30668448/39596935-83d61fb0-4ed9-11e8-8144-4430b2e77a52.png)

Package BIN Location:

![image](https://user-images.githubusercontent.com/30668448/39597024-cd6d3e9c-4ed9-11e8-818a-8085b200f786.png)

I did confirm that CoreNodeModels was set to Copy to Local. I made that modification to not copy and updated 1.3 references to 2.0 references. The same issue still exists. Here is what I see in Dynamo 2.0 when I load the package.

![image](https://user-images.githubusercontent.com/30668448/39597125-0dc655d2-4eda-11e8-9ced-6b53adc26ffd.png)

Thanks

**Comment by jnealb** [comment]
@kraftwerk15 thanks for the analysis and interesting suggestions. @Racel @kronz @smangarole fyi.

**Comment by mjkkirschner** [comment]
When a package path is added like this a notification is logged in the notification menu item and in the console.

**Comment by kraftwerk15** [comment]
But that would only be for _that_ session, correct??? 

I work with the console open, but that really wasn't until I started working with Zero Touch to learn more about what was going on. I would anticipate the overwhelmingly majority of Dynamo users do not even know there is a console. Unless I made note of that the moment that I loaded that .dll file, I would never know that the above issue is the reason the error was occurring.

And now that I think about it more, I will never be able to get to that information unless Dynamo can open when the error happens. The dialog box error helps, but the only fix is to delete the file at the said location, which could be suggested in the dialog box error.

@mjkkirschner That may be direct and I apologize. I just want to document this information so whenever someone on the internet comes crawling looking for why the same thing is happening to them, there is some type of answer to try.....without wasting hours like I did. :)
